### PR TITLE
Fix ParallelReadBuffer seek

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

https://github.com/ClickHouse/ClickHouse/pull/49539 forgot to assign `pos`. (I probably confused it with `nextImpl()`, where the caller sets `pos = working_buffer.begin()` automatically.)

### CODE REVIEW

**Review by Avogar — APPROVED** [review]

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 492e85eab28176254523657d4a419b9e5f765cb2 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50820/492e85eab28176254523657d4a419b9e5f765cb2/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by al13n321** [comment]
Added a test.

**Comment by alexey-milovidov** [comment]
Ok, but the tests are broken - need to fix them in a separate PR.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
LGTM, but the tests have failed. We cannot merge with failed tests.

File: tests/queries/0_stateless/02731_formats_s3.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+-- Tags: no-fasttest
+-- Tag no-fasttest: Depends on AWS
+
+-- Reading from s3 a parquet file of size between ~1 MB and ~2 MB was broken at some point
+-- (bug in CachedOnDiskReadBufferFromFile).
+insert into function s3(s3_conn, filename='test_02731_parquet.parquet') select cityHash64(number) from numbers(170000) settings s3_truncate_on_insert=1;
```
Comment (Avogar) [inline tests/queries/0_stateless/02731_formats_s3.sql:6]: We can add this file in folder `data_minio` and don't generate it all the time (and test will be able to run in parallel)

**Pull request merged**